### ISSUE TITLE
How to access others public resources. via client side API call now that client_id in URL is deprecated 

### ISSUE THREAD

**Issue by mc1903** [soundcloud/api#145]
I have been using [PSSoundCloud](https://github.com/Skeptech/PSSoundCloud) for a few years to access and download other users content via PowerShell, but this has now stopped working, since support for `client_id` in URL has been deprecated.

How do I access public resources, via the API, moving forward? Do I need to register an 'app' in SoundCloud (which I realise has been disabled for a long time now) and obtain OAuth _secrets_ or can I still use my `client_id` in the http header?

Your [API Guide](https://developers.soundcloud.com/docs/api/guide) mentions **Client-side Integrations** but the section the link references is missing. See the screengrabs below.

Can you please provide a client side authentication code example?

Many Thanks
M

![image](https://user-images.githubusercontent.com/13692287/135614882-841a1a09-e87f-4c2d-832d-98f23b7b1fde.png)

 
![image](https://user-images.githubusercontent.com/13692287/135615046-9430a778-12fb-449d-babb-d67c6bd7dd3c.png)

**Comment by ChrisPowellIinc** [comment]
Any luck on this?

**Comment by mc1903** [comment]
> Any luck on this?

Hi @ChrisPowellIinc. 

No, nothing as yet. but this issue was only opened at lunchtime on Friday and I am sure they don't work weekends. I am hoping to hear something this week.

I have a support ticket open too. Whatever comes from that I will update here.

M

**Comment by ChrisPowellIinc** [comment]
I saw this recently posted I think it can be modified to work: https://github.com/roncli/roncli.com/blob/6eb0937aaf1c15a2cbb4262f18e5356f7a3fa1dc/node-roncli-com/src/soundcloud/index.js

**Comment by mc1903** [comment]
Evening @ChrisPowellIinc,

I don't write in node.js, but looking at the _getAccessToken call_ it is `POST`ing the `grant_type` , `client_id` **and** `client_secret` in the http header into the `https://api.soundcloud.com/oauth2/token` endpoint to obtain an access/session token.

So I am pretty certain that it **will** require an _App_ to be registered against one's own SC account, which should then generate unique `client_id` and `client_secret` IDs.

If SC allow me to register an _App_ in my account, I will look to update/replace the PSSoundCloud code with the new OAuth authentication method. I have tinkered with RESTful API's for while now, so hopefully I can put something together.

I see that @rahul-sc has flagged this issue as a question, so I hope that we will get a response soon. I have heard nothing from my open support ticket today. :-(.

M

**Comment by ChrisPowellIinc** [comment]
Any known way to get the Soundcloud "client_secret" or was this only available with a registered Application?

**Comment by rahul-sc** [comment]
Thanks for your feedback @mc1903 Sorry that your usage experience is miserable. We'll consider publishing some information and clarity on our plans for the API.

**Comment by ChrisPowellIinc** [comment]
I was able to fetch player information like this and now I'm getting unauthorized anyway to resolve this? 

var client_id = "ca1f6b04464964bb9ed82eaa129f5cc7";
// Setup our new audio player class and pass it the playlist.
fetch(
  `https://api.soundcloud.com/playlists/1193693011?client_id=ca1f6b04464964bb9ed82eaa129f5cc7`
)
  .then((res) => res.json())
  .then((json) => {
    var playlist = json.tracks.map((track) => {
      return {
        id: track.id,
        title: track.title,
        src: track.stream_url + `?client_id=${client_id}`,
        artwork: track.artwork_url,
        duration: track.duration,
        genre: track.genre,
        howl: null,
      };
    });
    musicPlayer = new Player(playlist);
    musicPlayer.volume(0.15);
  });

**Comment by dasha-kobzeva** [comment]
Hello @ChrisPowellIinc,

Please check our blogpost regarding the auth changes: https://developers.soundcloud.com/blog/security-updates-api 

The usage of `client_id` was deprecated. A proper token has to be provided in every request in the `Authorisation` header. Otherwise the request is treated as unauthorised

**Comment by ChrisPowellIinc** [comment]
Yes, I saw the article. However registration for an app is closed and I can't seem to find working examples online.  Is there a demo you can point me to?

**Comment by dasha-kobzeva** [comment]
As I understand correctly you do not have an application? In that case unfortunately you cannot use our API until further notice when the registration is open. I apologize for the inconvenience.

**Comment by ChrisPowellIinc** [comment]
😕 can you PM me existing credentials? We customized a player using this api for an app.

**Comment by dasha-kobzeva** [comment]
@ChrisPowellIinc if you do not have an app, there are no credentials to give.

Could you please create a separate ticket and explain the application you are building? 
What endpoints you are calling, share some info about the player?

Thank you
